### PR TITLE
feat: add logging facility (SDKCF-4685)

### PR DESCRIPTION
**Pull request by leila-gu**
Added a Simple logging Utility class

### CODE REVIEW

File: sdk-utils/build.gradle [hunk 1]
```diff
@@ -21,7 +21,7 @@ dependencies {
 
   testImplementation 'junit:junit:4.12'
   testImplementation 'org.robolectric:robolectric:4.3'
-  testImplementation "org.amshove.kluent:kluent-android:1.57"
+  testImplementation "org.amshove.kluent:kluent-android:1.65"
```
Comment (vincentdelmundo) [inline sdk-utils/build.gradle:24]: nice!!

File: sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                break
+            }
+        }
+        return if (location != null) "${location.className}#${location.methodName}:${location.lineNumber} " else ""
+    }
+
+    /**
+     * Log messages as DEBUG.
+     */
+    fun debug(message: String?, vararg args: Any?) {
```
Comment (vincentdelmundo) [inline sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt]: Update Javadoc to all public functions.

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just two comments.
1. please see inline comment
2. please add usage/documentation in userguide.md and update changelong

File: sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt (hunk 2 above)
Comment (leila-gu) [reply sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt]: Updated

File: README.md [hunk 3]
```diff
@@ -60,10 +60,43 @@ The `AppInfo` class can be used to retrieve some properties of the App which nor
 ```kotlin
 val appName = AppInfo.instance.name
 val appVersion = AppInfo.instance.version
+```
+##  <a name="#logger-usage"></a>  Logging Utiliy
```
Comment (vincentdelmundo) [inline README.md]: Please fix spacing.

**Review by vincentdelmundo — APPROVED** [review]
LGTM. Just formatting comment, no need to request another review.

File: sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    /**
+     * Log messages as INFO.
+     * Will use [String.format] to format the message by substituting the arguments.
+     * @param cause: The given exception to log.
+     * @param message: The message to log.
+     * @param args: Arguments for the format string.
+     */
+    fun info(cause: Throwable? = null, message: String?, vararg args: Any?) {
```
Comment (MdUsmanGani) [inline sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt]: Do we need to keep both 'info' method with same 3 params with different sequence?

**Review by MdUsmanGani — APPROVED** [review]
Looks good to me, just one comment for confirmation.

File: README.md (hunk 3 above)
Comment (leila-gu) [reply README.md]: fixed

File: sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt (hunk 4 above)
Comment (leila-gu) [reply sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt]: Deleted the duplicated methods

**Pull request merged**